### PR TITLE
Implementing cache to delete application templates

### PR DESCRIPTION
**Pull request by somindatommy**
Resolves: https://github.com/wso2/product-is/issues/10155

### Proposed changes in this pull request
Implementing cache to delete application templates.

### CODE REVIEW

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java [hunk 1]
```diff
@@ -363,6 +365,43 @@ private Resource getResourceFromCacheByName(String resourceName, String tenantDo
...
+    }
+
+    private void removeResourceFromCache(int tenantId, String resourceId, String resourceName) {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                    .setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
```
Comment (IsuraD) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java:388]: why do we set tenant domain as super tenant and tenant id as relevant tenant id?

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java [hunk 2]
```diff
@@ -363,6 +365,43 @@ private Resource getResourceFromCacheByName(String resourceName, String tenantDo
...
+            ResourceByIdCacheKey resourceByIdCacheKey = new ResourceByIdCacheKey(resourceId, tenantDomain);
+            ResourceByNameCacheKey resourceByNameCacheKey = new ResourceByNameCacheKey(resourceName, tenantDomain);
+            ResourceCacheEntry resourceCacheEntry = resourceByIdCache.getValueFromCache(resourceByIdCacheKey);
+            if (resourceCacheEntry == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No cache entry found for resource: " + resourceId);
+                }
+                return;
```
Comment (IsuraD) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java:396]: don't we need to remove from resourceByNameCache?

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java (hunk 1 above)
Comment (somindatommy) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java]: Fixes in afaf1aa2548752b885a92a7ac5e783a8f7e02397

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java (hunk 2 above)
Comment (somindatommy) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/CachedBackedConfigurationDAO.java:383]: Fixed in afaf1aa2548752b885a92a7ac5e783a8f7e02397

**Review by IsuraD — APPROVED** [review]

**Comment by somindatommy** [comment]
Fixed by https://github.com/wso2/carbon-identity-framework/pull/3242

**Pull request closed without merging**